### PR TITLE
Map element datums

### PR DESCRIPTION
**Pull request by unid15**
- Added /datum/map_element objects. Vaults and away missions are children of them (as /datum/map_element/vault and /datum/map_element/away_mission, respectively)
- Merged "Jump to Vault" and "Jump to Away Mission" procs into "Jump to Map Element"
- Added "Load Map Element" proc that lets admins load any vault, away mission or external map file into the game. Away missions loaded through this proc function just like normal away missions - you can use the gateways and so on.
- Fixed maploader not processing underlying turfs correctly. Before: http://puu.sh/oYRuJ/8fd01b5f52.png . After: http://puu.sh/oYRGK/e93f1737ee.png . This should also fix some nasty runtimes

Jump to Map Element looks like this:

### CODE REVIEW

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -1104,6 +1105,69 @@ var/list/admin_verbs_mod = list(
...
+			if(x_coord == null) return
+
+			var/y_coord = input(usr, "Input the Y coordinate (X = [x_coord]): ", "Map element loading") as null|num
+			if(y_coord == null) return
+
+			var/z_coord = input(usr, "Input the Z coordinate. If it's higher than [world.maxz], a new Z-level will be created (X = [x_coord], Y = [y_coord]): ", "Map element loading") as null|num
+			if(z_coord == null) return
+
+			if(z_coord > world.maxz)
+				world.maxz++
```
Comment (PJB3005) [inline code/modules/admin/admin_verbs.dm]: Map loader does this by default if the z is larger than `world.maxz`

File: code/modules/admin/verbs/adminjump.dm [hunk 2]
```diff
@@ -110,60 +110,39 @@
...
 		return
 
 	if(config.allow_admin_jump)
 		var/list/vaults = list()
 
-		for(var/datum/vault/V in existing_vaults)
-			vaults["[V.map_name] at [V.location ? "[V.location.x], [V.location.y], [V.location.z]" : "UNKNOWN"]"] = V
+		for(var/datum/map_element/V in map_elements)
+			var/name = "[V.name ? V.name : V.file_path] at [V.location ? "[V.location.x], [V.location.y], [V.location.z]" : "UNKNOWN"]"
+			if(istype(V, /datum/map_element/vault))
```
Comment (PJB3005) [inline code/modules/admin/verbs/adminjump.dm]: Hardcode!

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -1104,6 +1105,68 @@ var/list/admin_verbs_mod = list(
...
+
+			z_coord = input(usr, "Input the Z coordinate. If it's higher than [world.maxz], a new Z-level will be created (X = [x_coord], Y = [y_coord]): ", "Map element loading") as null|num
+			if(z_coord == null) return
+
+		if("Cancel")
+			return
+
+	log_admin("[key_name(src)] is loading [ME.file_path] at [x_coord], [y_coord], [z_coord]")
+	message_admins("[key_name_admin(src)] is loading [ME.file_path] at [x_coord], [y_coord], [z_coord]")
+	ME.load(x_coord, y_coord, z_coord)
```
Comment (clusterfack) [inline code/modules/admin/admin_verbs.dm]: My only question is shouldn't this include some kind of sanity/warning for when a loaded maps width/height are too large and will cause it to collide with the edge of the map or another area? 

It seems like that should be possible

Comment (unid15) [inline code/modules/admin/admin_verbs.dm:1167]: There is no proc to get the maps dimensions, and the map loader should extend the map automatically anyways if that is the case

Comment (clusterfack) [inline code/modules/admin/admin_verbs.dm]: I think in the future the map loader needs to be given that kind of sanity

**Pull request merged**